paging-runtime-composeui/src/androidMain/kotlin/app/cash/paging/PagingPlaceholderKey.kt

```diff
@@ -0,0 +1,35 @@
+package app.cash.paging
+
+import android.annotation.SuppressLint
+import android.os.Parcel
+
+private typealias Creator<T> = android.os.Parcelable.Creator<T>
+
+actual fun createPlaceholderKey(index: Int): PagingPlaceholderKey =
+  RealPagingPlaceholderKey(index)
+
+@SuppressLint("BanParcelableUsage")
+private data class RealPagingPlaceholderKey(
```

Comment:
Basically a copy [from source](https://github.com/androidx/androidx/blob/androidx-main/paging/paging-compose/src/main/java/androidx/paging/compose/LazyPagingItems.kt#L360-L380)

Reply:
yup, just copied over from source but I'll replace the boilerplate with the annotation. 

One thing I had to work around though is given - 
```
abstract class PagingPlaceholderKey : Parcelable
```

I still have to explicitly extend `Parcelable` from the subclass when applying the annotation. 
i.e
```
// Compiler complains "No Parcelable supertype", even though  PagingPlaceholderKey extends Parcelable
@Parcelize
data class RealPagingPlaceholderKey(
  private val index: Int
) : PagingPlaceholderKey(), Parcelable
```

however this works
```
@Parcelize
data class RealPagingPlaceholderKey(
  private val index: Int
) : PagingPlaceholderKey(), Parcelable
```